ledger/cow.go

```diff
@@ -198,8 +202,9 @@ func (cb *roundCowState) trackCreatable(creatableIndex basics.CreatableIndex) {
 	cb.trackedCreatables[cb.groupIdx] = creatableIndex
 }
 
-func (cb *roundCowState) addTx(txn transactions.Transaction, txid transactions.Txid) {
+func (cb *roundCowState) addTx(txn transactions.Transaction, txid transactions.Txid, inners uint64) {
 	cb.mods.Txids[txid] = txn.LastValid
+	cb.txnCount += 1 + inners
```

Comment:
Currently, this code makes use of the uniqueness of a map and silently ignores duplicate transactions. With your change that is no longer the case.

It's very possible that we depend on that property.

I ran into this property recently during some Indexer testing. My synthetic transactions weren't random enough and if the block size was large enough I would create duplicate transactions. The Eval function (with validation disabled) would silently ignore them, which I only noticed because of unexpected TxnCounter values.

Reply:
I'm not sure I understand.  Do you mean for dupe detection?  Inner transactions are allowed to be duplicates.  There's no need for replay protection, because these transactions are not signed.  They ran because the parent ran (and that is protected). So they can't be replayed.

Reply:
Yes, this is related to dupe detection. The existing code correctly handles duplicate transactions and the new code does not.

Reply:
Meaning that the change does not detect "inner dupes", right?  That was the intent.  If you also think this changes behavior for top-level, that would be bad.

Reply:
I believe @winder is saying this function may be called multiple times with the same txn, and the behavior before this PR could properly handle this, since it has no side effects besides putting things into maps. If this is a property that we need to maintain (seems strange to me), then perhaps we could have a new map from txID to inner txn count instead of the monotonic `txnCount` variable.

Reply:
Yeah, we had a chat.  You are exactly right, and so is he. I intend to figure out if use that property, but I suspect not.  Will ran into it because he fed the validator synthetic blocks that had duplicates in them.  But we should never allow such things in a real block.